include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -2319,11 +2365,13 @@ __parallel_reduce_by_segment_fallback(oneapi::dpl::__internal::__device_backend_
                                       _BinaryOperator __binary_op,
                                       /*known_identity=*/std::false_type)
 {
+    const auto __n = __keys.size();
+    assert(__n > 0);
+
```

Comment:
Optional suggestion: roll back these changes, since they are not related to this PR.

Reply:
"not related" but very useful. As far as we are changing  the file, lets keep the minor change.